lightblue-migration-monitor/src/main/java/com/redhat/lightblue/migrator/monitor/Monitor.java

```diff
@@ -0,0 +1,93 @@
+import com.redhat.lightblue.client.LightblueException;
+import com.redhat.lightblue.client.Literal;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.response.LightblueDataResponse;
+import com.redhat.lightblue.migrator.MigrationConfiguration;
+import com.redhat.lightblue.migrator.MigrationJob;
+
+public class Monitor {
+
+    private final LightblueClient lightblueClient;
+
+    public Monitor(MonitorConfiguration cfg) {
+        if (cfg.getClientConfig() != null) {
+            lightblueClient = new LightblueHttpClient(cfg.getClientConfig());
+        } else {
+            lightblueClient = new LightblueHttpClient();
+        }
+    }
+
+    public void runCheck(final Notifier... notifiers) throws LightblueException {
+        List<String> configurationsMissingJobs = new ArrayList<>();
+        for (MigrationConfiguration cfg : findMigrationConfigurations()) {
+            long period = parsePeriod(cfg.getPeriod());
+            Date endDate = new Date();
+            Date startDate = new Date(endDate.getTime() - period);
+
+            int jobs = countMigrationJobs(cfg.getConfigurationName(), startDate, endDate);
+            if (jobs <= 0) {
+                configurationsMissingJobs.add(cfg.getConfigurationName());
+            }
+        }
+
+        if (configurationsMissingJobs.isEmpty()) {
+            for (Notifier n : notifiers) {
+                n.sendSuccess();
+            }
+        } else {
+            for (Notifier n : notifiers) {
+                n.sendFailure(configurationsMissingJobs);
+            }
+        }
+    }
+
+    /**
+     * Returns the period in msecs
+     */
+    private static long parsePeriod(String periodStr) {
+        PeriodFormatter fmt = PeriodFormat.getDefault();
+        Period p = fmt.parsePeriod(periodStr);
+        return p.toStandardDuration().getMillis();
+    }
+
+    private MigrationConfiguration[] findMigrationConfigurations() throws LightblueException {
+        DataFindRequest findConfigurations = new DataFindRequest(MigrationConfiguration.ENTITY_NAME);
+        findConfigurations.where(Query.withValue("period", Query.neq,
+                Literal.pojo(null)));
+        findConfigurations.select(
+                Projection.includeField("configurationName"),
+                Projection.includeField("period"));
+
+        return lightblueClient.data(findConfigurations, MigrationConfiguration[].class);
+    }
+
+    private int countMigrationJobs(String configurationName, Date startDate, Date endDate) throws LightblueException {
+        DataFindRequest findJobs = new DataFindRequest(MigrationJob.ENTITY_NAME);
+        findJobs.where(
+            Query.and(
+                Query.withValue("configurationName", Query.eq, configurationName),
```

Comment:
I had actually thought about the monitor possibly checking a period or two behind just to provide some slack. This would be fine for jobs that generate every hour, but dailies would become problematic.